Makefile

```diff
@@ -54,38 +59,46 @@ prepare_upload_data: build_lambda_binary                     ## Create archive t
 	aws s3 sync build/s3/ s3://$(BUCKET_NAME)/
 
-vet-check:                                                   ## Verify vet compliance
-ifeq ($(shell go tool vet -all -shadow=true $(GO_FILES) 2>&1 | wc -l), 0)
+vet-check:                                                                   ## Verify vet compliance
+ifeq ($(shell go tool vet -all -shadow=true . 2>&1 | wc -l), 0)
 	@printf "ok\tall files passed go vet\n"
 else
 	@printf "error\tsome files did not pass go vet\n"
-	@go tool vet -all -shadow=true $(GO_FILES) 2>&1
+	@go tool vet -all -shadow=true . 2>&1
 endif
 
-fmt-check:                                                   ## Verify fmt compliance
+fmt-check:                                                                   ## Verify fmt compliance
 ifneq ($(shell gofmt -l -s $(GO_FILES) | wc -l), 0)
-	@printf "error\tsome files did not pass go fmt: $(shell gofmt -l -s $(GO_FILES))\n"; exit 1
+	@printf "error\tsome files did not pass go fmt, fix the following formatting diff\n"
+	@gofmt -l -s -d $(GO_FILES)
+	@exit 1
 else
 	@printf "ok\tall files passed go fmt\n"
 endif
 
-test:                                                        ## Test go code and coverage
+test:                                                                        ## Test go code and coverage
 	@go test -covermode=count -coverprofile=$(COVER_PROFILE) $(BINARY_PKG)
 
-full-test: test fmt-check vet-check                          ## Pass test / fmt / vet
+full-test: fmt-check vet-check test                                          ## Pass test / fmt / vet
 
-html-cover: test                                             ## Display coverage in HTML
+html-cover: test                                                             ## Display coverage in HTML
 	@go tool cover -html=$(COVER_PROFILE)
 
-help:                                                        ## Show this help
+travisci-cover: html-cover                                                   ## Generate coverage in the TravisCI format, fails unless executed from TravisCI
+	@goveralls -coverprofile=$(COVER_PROFILE) -service=travis-ci
+
+travisci-checks: fmt-check vet-check                                         ## Pass fmt / vet and calculate test coverage
+
+travisci: prepare_bindata travisci-checks prepare_upload_data travisci-cover ## Executed by TravisCI
+
+help:                                                                        ## Show this help
 	@printf "Rules:\n"
 	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//'
 
-.PHONY: all clean check_dep bindata build_lambda_binary prepare_upload_data build_local upload vet-check fmt-check
-.PHONY: test full-test html-cover help
+.PHONY: all clean check_dep bindata build_lambda_binary prepare_upload_data build_local upload vet-check fmt-check check_deps prepare_bindata test full-test html-cover help travisci-checks travisci-cover travisci
```

Comment:
Notes:
* bindata rule doesn't seem to exist anymore - could be removed from build_local / build_lambda_binary
* the check_dep rule can be removed - as renamed check_deps